plugins/follow_me/follow_me_impl.cpp

```diff
@@ -0,0 +1,294 @@
+#include "follow_me_impl.h"
+#include "device_impl.h"
+#include "global_include.h"
+#include "px4_custom_mode.h"
+#include <functional>
+#include <cmath>
+#include <array>
+
+namespace dronecore {
+
+using namespace std::placeholders; // for `_1`
+
+FollowMeImpl::FollowMeImpl() :
+    PluginImplBase(),
+    _time(),
+    _timeout_cookie(nullptr),
+    _motion_report(),
+    _motion_report_timer(_time),
+    _estimatation_capabilities(0),
+    _config(),
+    _followme_mode_active(false)
+{
+}
+
+FollowMeImpl::~FollowMeImpl()
+{
+}
+
+void FollowMeImpl::init()
+{
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_HEARTBEAT,
+        std::bind(&FollowMeImpl::process_heartbeat, this, _1), static_cast<void *>(this));
+}
+
+void FollowMeImpl::deinit()
+{
+    _parent->unregister_all_mavlink_message_handlers(this);
+}
+
+void FollowMeImpl::timeout_occurred()
+{
+#ifdef FOLLOW_ME_TESTING
```

Comment:
Testing should be done using the public API. This internal testing doesn't make sense in my opinion. The point of a unit test is to test the API, so it needs to access it from the outside.

Reply:
That is a bit of a challenge currently - somehow going to have to test the public API while mocking movement of the GCS.

Reply:
As of now we can test from PUBLIC API `void FollowMe::set_follow_info(const FollowInfo &info)`